### PR TITLE
Implement functionality that allows users to filter stores by any number of items.

### PR DESCRIPTION
**Pull request by AnudeepYakkala**
Added a filter object containing all of the user's selected items as options. The user is able to select any number of filters and only the stores that contain all the selected filter items are displayed. 

I added sample data to test the filter and removed it before uploading my code. The sample data had 1 store with milk, eggs, and water and another store with water, apples and, toothpaste. The following is a gif showing the new functionality with sample data:
Filter

### CODE REVIEW

File: capstone/client/src/components/StoresPage.js [hunk 1]
```diff
@@ -44,12 +46,9 @@ class StorePage extends Component {
...
           longitude : this.props.store.state.longitude,
-        }))
-    }
-
-    componentDidMount = () => {
-      // Get Stores from database.
-      this.getStores();
+        }));
+        // Get Stores from database.
+        this.getStores();
```
Comment (chen-dawn) [inline capstone/client/src/components/StoresPage.js:51]: I think async requests should go in `componentDidiMount`, where we can call `setState`. Maybe that's why we have the UI lag on the front page. See
https://medium.com/all-about-react/react-basic-constructor-componentdidmount-and-render-9476f8d28f0f
https://stackoverflow.com/questions/55182526/fetching-data-in-constructor

File: capstone/client/src/components/FilterStores.js [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+    selectedFilters.delete(event.target.innerText);
+    const stores = this.filterStores(selectedFilters);
+    this.setState({
+      selectedFilters,
+    });
+    this.props.onFilterChange(stores);
+  }
+
+  filterStores(selectedFilters) {
+    const stores = this.state.originalStores.filter(store => {
```
Comment (luli0521) [inline capstone/client/src/components/FilterStores.js:56]: nit: add parentheses for 'store'

File: capstone/client/src/components/FilterStores.js [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+      const storeItems = new Set();
+      store.items.forEach((item) => storeItems.add(item.type));
+      let containsAll = true;
+      selectedFilters.forEach((item) => {
+        if (!storeItems.has(item)) {
+          containsAll = false
+        }
```
Comment (luli0521) [inline capstone/client/src/components/FilterStores.js:63]: containsAll = storeItems.has(item);

Comment (AnudeepYakkala) [reply capstone/client/src/components/FilterStores.js:63]: I am trying to check if the store is missing any of the items from the selected filter items. Using "containsAll = storeItems.has(item)" would overwrite the value of containsAll to true if a store is missing 1 item but contains the item that is checked next.

**Pull request merged**